app/multieditmanager.cpp

```diff
@@ -152,6 +153,39 @@ FeatureLayerPair MultiEditManager::editableFeature()
   return FeatureLayerPair( oneFeature, mTempLayer.get() );
 }
 
+void MultiEditManager::deleteSelectedFeature()
+{
+  if ( !mModel || mModel->count() == 0 || !mLayer )
+    return;
+
+  const FeatureLayerPairs pairs = mModel->features();
+  QgsFeatureIds fids;
+  fids.reserve( pairs.count() );
+  for ( const FeatureLayerPair &pair : pairs )
+  {
+    fids.insert( pair.feature().id() );
+  }
```

Comment:
I think we could save a bit of memory if we read the data directly from the model instead of copying them into `pairs`. Something like this should work in theory:

```suggestion

  QgsFeatureIds fids;
  fids.reserve( mModel->rowCount() );
  for ( int i = 0; mModel->rowCount(); i++ )
  {
    fids.insert( mModel->data( index( i ), FeaturesModel::FeatureId );
  }
```

It's quite a shame that we need to extract feature IDs in the first place. I guess for future it would be better if the model only included feature Ids and requested additional attributes when needed. Something like what @uclaros started to do in https://github.com/MerginMaps/mobile/pull/4035